src/main/java/mate/academy/springboot/datajpa/service/mapper/ProductMapper.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.springboot.datajpa.service.mapper;
+
+import mate.academy.springboot.datajpa.dto.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductMapper implements RequestDtoMapper<ProductRequestDto, Product>,
+        ResponseDtoMapper<ProductResponseDto, Product> {
+    @Override
+    public Product mapToModel(ProductRequestDto dto) {
+        Product product = new Product();
+        product.setTitle(dto.getTitle());
+        product.setPrice(dto.getPrice());
+        Category category = new Category();
+        category.setId(dto.getCategoryId());
+        product.setCategory(category);
+        return product;
+    }
+
+    @Override
+    public ProductResponseDto mapToDto(Product product) {
+        ProductResponseDto productResponseDto = new ProductResponseDto();
+        productResponseDto.setId(product.getId());
+        productResponseDto.setTitle(product.getTitle());
+        productResponseDto.setPrice(product.getPrice());
+        productResponseDto.setCategoryId(product.getId());
```

Comment:
seems wrong
and how you will be able to set categoryId without LazyInitializationException?
are you sure you need to lazy-fetch category?